### PR TITLE
migrate test runs from travis to github actions

### PR DESCRIPTION
**Pull request by falkenhawk**
also includes php 8.0 and php 8.1, with allowance to fail

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Tests
+
+on:
```
Comment (partikus) [inline .github/workflows/tests.yml:3]: I would change it to:
```yaml
on:
  pull_request: ~
  push:
    branches:
      - "master"
```

**Review by partikus — APPROVED** [review]
looks good 👍 one more thing I would change

**Comment by falkenhawk** [comment]
@marcing @partikus I have no idea why it fails on php 5.x. On travis it worked fine https://app.travis-ci.com/github/ovos/doctrine1/jobs/513037759

I cannot spend more time on this, I am leaving this for you

**Comment by partikus** [comment]
:wink:

doctrine1 – 1674TestCase php 2022-08-09 06-20-02 

 I believe for PHP 5.3 we should use:
 
```php
if (PHP_VERSION_ID >= 50300) {
            $this->assertEqual($xml, '<?xml version="1.0" encoding="utf-8"?>
<data><User id="4" type="0" email_id="1"><name>zYne</name><loginname/><password/><created/><updated/><Phonenumber><Phonenumber id="2" entity_id="4"><phonenumber>123 123</phonenumber><Entity/></Phonenumber></Phonenumber></User></data>
');
        } else {
            $this->assertEqual($xml, '<?xml version="1.0" encoding="utf-8"?>
<data><User id="4" type="0" email_id="1"><name>zYne</name><loginname></loginname><password></password><created></created><updated></updated><Phonenumber><Phonenumber id="2" entity_id="4"><phonenumber>123 123</phonenumber><Entity></Entity></Phonenumber></Phonenumber></User></data>
');
        }
```

but then, the bugfix does not make sense 😮

**Pull request merged**